src/NUnitFramework/tests/Internal/Results/TestResultApiTests.cs

```diff
@@ -75,5 +76,55 @@ public void DoesNotThrowForMissingInnerException()
                 method.Invoke(result, exception);
             }
         }
+
+        /// <summary>
+        /// This models a .NET Framework race condition that resulted in stack traces throwing AccessViolationException
+        /// which would produce another AccessViolationException on attempting to print its stack trace in turn.
+        /// </summary>
+        [Test]
+        public void RecordingExceptionDoesNotThrowWhenExceptionMembersThrowRecursively()
+        {
+            foreach (var method in RecordExceptionMethods)
+            foreach (var result in TestResults)
+            {
+                method.Invoke(result, new BrokenException());
+
+                Assert.That(result, Has.Property("Message").StartWith(
+                    "NUnit.Framework.Internal.Results.TestResultApiTests+BrokenException : BrokenException was thrown by the Exception.Message property." + Environment.NewLine
+                    + "BrokenException was thrown by the Exception.Data property."));
+
+                Assert.That(result, Has.Property("StackTrace").StartWith(
+                    "BrokenException was thrown by the Exception.StackTrace property."));
+            }
+        }
+
+        private sealed class BrokenException : Exception
+        {
+            public override string Message => throw this;
```

Comment:
I couldn't figure out which was more evil, `throw this` or `throw new BrokenException();`

Both certainly got the point across 😁

![image](https://user-images.githubusercontent.com/8040367/60559320-13e50f00-9d1a-11e9-813a-02067d657b0f.png)

ReSharper: "The active test run was aborted. Reason: test host process crashed."